dask_geopandas/io/file.py

```diff
@@ -0,0 +1,121 @@
+
+    def project_columns(self, columns):
+        """Return a new FileFunctionWrapper object with
+        a sub-column projection.
+        """
+        if columns == self.columns:
+            return self
+        return FileFunctionWrapper(self.layer, columns)
+
+    def __call__(self, part):
+        path, row_offset, batch_size = part
+
+        import pyogrio
+
+        df = pyogrio.read_dataframe(
+            path,
+            layer=self.layer,
+            columns=self.columns,
+            read_geometry=self.read_geometry,
+            skip_features=row_offset,
+            max_features=batch_size,
+        )
+        return df
+
+
+def read_file(
+    path, npartitions=None, chunksize=None, layer=None, columns=None, **kwargs
+):
+    """
+    Read a GIS file into a Dask GeoDataFrame.
+
+    This function requires `pyogrio <https://github.com/geopandas/pyogrio/>`__.
+
+    Parameters
+    ----------
+    path : str
+        The absolute or relative path to the file or URL to
+        be opened.
+    npartitions : int, optional
+        The number of partitions to create. Either this or `chunksize` should
+        be specified.
+    chunksize : int, optional
+        The number of rows per partition to use. Either this or `npartitions`
+        should be specified.
+    layer : int or str, optional (default: first layer)
+        If an integer is provided, it corresponds to the index of the layer
+        with the data source.  If a string is provided, it must match the name
+        of the layer in the data source.  Defaults to first layer in data source.
+    columns : list-like, optional (default: all columns)
+        List of column names to import from the data source.  Column names must
+        exactly match the names in the data source, and will be returned in
+        the order they occur in the data source.  To avoid reading any columns,
+        pass an empty list-like.
+
+    """
+    import pyogrio
+    from dask.layers import DataFrameIOLayer
+
+    # TODO smart inference for a good default partition size ?
+    if (npartitions is None) == (chunksize is None):
+        raise ValueError("Exactly one of npartitions and chunksize must be specified.")
+
+    if "skip_features" in kwargs or "max_features" in kwargs:
+        # TODO we currently use those keywords already for reading in each
+        # partition (we would need to take those into account for determining
+        # the part start/ends)
+        raise ValueError(
+            "The 'skip_features'/'max_feature' keywords are not yet supported"
+        )
+    if kwargs:
+        raise ValueError("Additional pyogrio keywords are not yet supported")
+
+    total_size = pyogrio.read_info(path, layer=layer)["features"]
+
+    if chunksize is None:
+        chunksize = int(ceil(total_size / npartitions))
+
+    # TODO this could be inferred from read_info ?
+    meta = pyogrio.read_dataframe(path, layer=layer, columns=columns, max_features=5)
```

Comment:
With https://github.com/geopandas/pyogrio/pull/30 in, we can maybe cut pyogrio 0.4.0 in a few days and create meta from `read_info`?

Reply:
crossref: [pyogrio 0.4.0 planning](https://github.com/geopandas/pyogrio/issues/37)

Reply:
On the other hand, if using the dtypes from `read_info`, we would still need to create an empty DataFrame with those dtypes. Which gives some additional code complexity. If we think this small read will generally not be costly / a problem, then the above might actually be the easiest.

Reply:
Do you even need to read 5 rows?  Would 1 suffice?

If you end up needing a DataFrame regardless, then having pyogrio return that seems best.

We may also want to consider allowing you to set a meaningful `max_features=0` to produce an empty GeoDataFrame with the correct columns and types.  Right now since we're using an unigned int for that, 0 is equivalent to not set => read all.